lib/crawler/keyword/schemas/keyword_csv_file.ex

```diff
@@ -0,0 +1,27 @@
+defmodule Crawler.Keyword.Schemas.KeywordCSVFile do
+  use Ecto.Schema
+
+  import CrawlerWeb.Gettext
+  import Ecto.Changeset
+
+  embedded_schema do
+    field :file, :map
+  end
+
+  def create_changeset(keyword_file, attrs \\ %{}) do
+    keyword_file
+    |> cast(attrs, [:file])
+    |> validate_required([:file])
+    |> validate_file_type()
+  end
+
+  defp validate_file_type(changeset) do
+    validate_change(changeset, :file, fn :file, file ->
+      if Path.extname(file.filename) == ".csv" and file.content_type == "text/csv" do
+        []
+      else
+        [file: gettext("Invalid CSV file")]
```

Comment:
How about
```suggestion
        [file: gettext("Uploaded file is not a valid CSV file")]
```

Reply:
Fixed in [71537f0](https://github.com/vnntsu/elixir-ic/pull/36/commits/71537f0e5eb384be7d8ce4bd12a48e3b9856bc83)